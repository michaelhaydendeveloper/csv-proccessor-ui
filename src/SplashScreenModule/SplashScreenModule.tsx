import {Component} from "react";
import {
    Text,
    TouchableHighlight,
    View,
    StyleSheet,
    Modal
} from "react-native";
import Default from "./intl/default";


class SplashScreenModule extends Component<any, any> {
    state = {
        modalVisible: false
    };

    setModalVisible = (visible: boolean) => {
        this.setState({ modalVisible: visible });
    }

    onRequestCloseHandler = () => {
        this.setState({ modalVisible: false });
    }

    render() {
        const { modalVisible } = this.state;
        const defaultIntl = new Default();
        const welcomeTitle = defaultIntl.getSectionMessage('welcomeTitle');
        const welcomeMessage = defaultIntl.getSectionMessage('welcomeMessage');
        const hideModalButtonLabel = defaultIntl.getSectionMessage('hideModalButtonLabel');
        const showModalButtonLabel = defaultIntl.getSectionMessage('showModalButtonLabel');
        return (
            <View style={styles.centeredView}>
                <Modal
                    animationType='slide'
                    transparent={true}
                    visible={modalVisible}
                    onRequestClose={this.onRequestCloseHandler}
                >
                    <View style={styles.centeredView}>
                        <View style={styles.modalView}>
                            <Text style={styles.welcome}>{welcomeTitle}</Text>
                            <Text style={styles.instruction}>
                                {welcomeMessage}
                            </Text>
                            <TouchableHighlight
                                style={styles.openButton}
                                onPress={() => {
                                    this.setModalVisible(!modalVisible);
                                }}
                            >
                                <Text style={styles.textStyle}>{hideModalButtonLabel}</Text>
                            </TouchableHighlight>
                        </View>
                    </View>
                </Modal>
                <TouchableHighlight
                    style={styles.openButton}
                    onPress={() => {
                        this.setModalVisible(true);
                    }}
                >
                    <Text style={styles.textStyle}>{showModalButtonLabel}</Text>
                </TouchableHighlight>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        backgroundColor: '#f5fcff',
        justifyContent: 'center',
        padding: 10,
    },
    instruction: {
        color: '#333333',
        marginBottom: 5,
        textAlign: 'center',
    },
    welcome: {
        fontSize: 20,
        margin: 10,
        textAlign: 'center',
        padding: 10,
    },
    centeredView: {
        justifyContent: "center",
        alignItems: "center",
        marginTop: 22,
        padding: 10,
    },
    modalView: {
        margin: 20,
        backgroundColor: "white",
        borderRadius: 20,
        padding: 35,
        alignItems: "center",
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5
    },
    openButton: {
        backgroundColor: "#F194FF",
        borderRadius: 20,
        padding: 10,
        elevation: 2
    },
    textStyle: {
        color: "white",
        fontWeight: "bold",
        textAlign: "center",
    },
});

export default SplashScreenModule;