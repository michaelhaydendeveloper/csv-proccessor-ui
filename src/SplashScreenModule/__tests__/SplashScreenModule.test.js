import React from 'react';
import renderer from 'react-test-renderer';
import SplashScreenModule from '../SplashScreenModule';

test('renders correctly', () => {
    const tree = renderer.create(<SplashScreenModule />).toJSON();
    expect(tree).toMatchSnapshot();
});
