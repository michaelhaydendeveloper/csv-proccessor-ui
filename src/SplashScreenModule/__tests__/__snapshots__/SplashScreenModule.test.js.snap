// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 22,
      "padding": 10,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 22,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "margin": 10,
              "padding": 10,
              "textAlign": "center",
            }
          }
        >
          Welcome Message Here
        </Text>
        <Text
          style={
            Object {
              "color": "#333333",
              "marginBottom": 5,
              "textAlign": "center",
            }
          }
        >
          Welcome Message Here
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#F194FF",
              "borderRadius": 20,
              "elevation": 2,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Hide Modal
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F194FF",
        "borderRadius": 20,
        "elevation": 2,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Show Modal
    </Text>
  </View>
</View>
`;
